tools/test-fwe.sh

```diff
@@ -26,10 +33,17 @@ parse_args "$@"
 
 if ${WITH_ROS2_SUPPORT}; then
     source /opt/ros/galactic/setup.bash
-    # colcon hides the test output, so use tail in the background.
-    tail -F log/latest_test/iotfleetwise/stdout_stderr.log &
-    colcon test --return-code-on-test-failure
+    cd build/iotfleetwise
+    if [ -n "${IGNORE_TESTS}" ]; then
+        ctest -V ${IGNORE_TESTS}
+    else
+        ctest -V
+    fi
 else
     cd build
-    ctest -V
+    if [ -n "${IGNORE_TESTS}" ]; then
+        ctest -V ${IGNORE_TESTS}
+    else
+        ctest -V
+    fi
 fi
```

Comment:
You don't need to repeat the command 4 times. You can run just `source` inside the if and run ctest once since the command is now the same for both:

```
if ${WITH_ROS2_SUPPORT}; then
    source /opt/ros/galactic/setup.bash
fi

cd build
ctest -V ${IGNORE_TESTS}

Reply:
Done